baselines/fedpara/fedpara/main.py

```diff
@@ -0,0 +1,135 @@
+"""Main script for running FedPara."""
+
+import flwr as fl
+import hydra
+from flwr.common.typing import Config
+from hydra.core.hydra_config import HydraConfig
+from hydra.utils import instantiate
+from omegaconf import DictConfig, OmegaConf
+
+from fedpara import client, server, utils
+from fedpara.dataset import load_datasets
+from fedpara.utils import (
+    get_parameters,
+    save_results_as_pickle,
+    seed_everything,
+    set_client_state_save_path,
+)
+
+
+@hydra.main(config_path="conf", config_name="mnist", version_base=None)
+def main(cfg: DictConfig) -> None:
+    """Run the baseline.
+
+    Parameters
+    ----------
+    cfg : DictConfig
+        An omegaconf object that stores the hydra config.
+    """
+    # 1. Print parsed config
+    print(OmegaConf.to_yaml(cfg))
+    seed_everything(cfg.seed)
+    OmegaConf.to_container(cfg, resolve=True)
+    if "state_path" in cfg:
+        state_path = set_client_state_save_path(cfg.state_path)
+    else:
+        state_path = None
+
+    # 2. Prepare dataset
+    train_loaders, test_loader = load_datasets(
+        config=cfg.dataset_config,
+        num_clients=cfg.num_clients,
+        batch_size=cfg.batch_size,
+    )
+
+    # 3. Define clients
+    # In this scheme the responsability of choosing the client is on the client manager
+
+    client_fn = client.gen_client_fn(
+        train_loaders=train_loaders,
+        test_loader=test_loader,
+        model=cfg.model,
+        num_epochs=cfg.num_epochs,
+        args={"device": cfg.client_device, "algorithm": cfg.algorithm},
```

Comment:
```suggestion
        args={"algorithm": cfg.algorithm},
```

Reply:
it's better to not set the device of clients directly in the config. Instead, based on `client_resources.num_gpus` it will be set to use CUDA (if a GPU is available). This is how other baselines do it (e.g. in [FedProx](https://github.com/adap/flower/blob/d7be8fb64507a83968479782976b9458435f909a/baselines/fedprox/fedprox/client.py#L155). See my comment later in `get_client_fn`

Reply:
Also, `algorithm` is not found in all configs (e.g. in cifar10)... maybe this is just some on-going work for MNIST? I think the easy fix will be to do in `cifar10.yaml` and `cifar100.yaml` the same as you do in `mnist.yaml` regarding the algorithm